New tab page search box

tomasstrba opened the pull request:
<!--
Note: This template is a reminder of our Engineering Expectations and Definition of Done. Remove sections that don't apply to your changes.

⚠️ If you're an external contributor, please file an issue before working on a PR. Discussing your changes beforehand will help ensure they align with our roadmap and that your time is well spent.
-->

Task/Issue URL: https://app.asana.com/1/137249556945/project/1148564399326804/task/1210625630564796?focus=true
Tech Design URL: -
CC:

### Description
_TBD_
### Testing Steps
<!-- Assume the reviewer is unfamiliar with this part of the app -->
1. _TBD_
2.

### Impact and Risks
<!-- 
What's the impact on users if something goes wrong?

High: Could affect user privacy, lose user data, break core functionality
Medium: Could disrupt specific features or user flows
Low: Minor visual changes, small bug fixes, improvement to existing features
None: Internal tooling, documentation
-->

#### What could go wrong?
<!-- Describe specific scenarios and how you've addressed them -->

### Quality Considerations
<!-- 
Focus on what matters for your changes:
- What edge cases exist?
- How does this affect performance?
- What monitoring have you added?
- What documentation needs updating?
- How does this impact privacy/security?
-->

### Notes to Reviewer
<!-- Anything specific you want reviewers to focus on -->

---
###### Internal references:
[Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f) | [Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552) | [Tech Design Template](https://app.asana.com/0/59792373528535/184709971311943)

tomasstrba commented on macOS/DuckDuckGo/NavigationBar/View/AddressBarTextField.swift:
```diff
@@ -1280,3 +1240,51 @@ extension AddressBarTextField: SuggestionViewControllerDelegate {
+            let result = await NSApp.delegateTyped.privacyFeatures.httpsUpgrade.upgrade(url: url)
+            switch result {
+            case let .success(upgradedUrl):
+                DispatchQueue.main.async {
```
There was also a bug where the completion handler was being called from a background thread after upgrading the URL.

ayoy commented on macOS/LocalPackages/NewTabPage/Sources/NewTabPage/Omnibar/NewTabPageDataModel+Omnibar.swift:
```diff
@@ -0,0 +1,196 @@
+
+public extension NewTabPageDataModel {
+
+    // https://github.com/duckduckgo/content-scope-scripts/blob/cb36865bf4087456dd3da282bf49ccb59cbfcc47/special-pages/pages/new-tab/app/omnibar/omnibar.md
```
Good idea to link documentation in these files!

Here’s the HTML docs link that will always be up-to-date: https://duckduckgo.github.io/content-scope-scripts/documents/New_Tab_Page.Omnibar_Widget.html

tomasstrba commented on macOS/DuckDuckGo/NewTabPage/NewTabPageOmnibarSuggestionsProvider.swift:
```diff
@@ -0,0 +1,91 @@
+
+private extension SuggestionResult {
+
+    var asNewTabPageSuggestions: NewTabPageDataModel.Suggestions {
```
Same as above

tomasstrba commented:
@ayoy, I’ve addressed your comments and the PR is ready for another review. Perhaps we could still get this into the internal build today.

Just a quick note: I’m planning to merge this into the main branch first, then cherry-pick it into the release branch. The diff gets pretty messy if I set the release branch as the base.

Also, a heads-up that I might add a few more commits in the meantime if anything is blocking tests or other checks.

ayoy reviewed (approved):
LGTM! Thanks for addressing the feedback!
